Merge MIP 203

rdenarie opened the pull request:
Add css helper for full window height and width

<!-- Ensure to provide github issue and task id in the title -->
<!-- Choose between feat and fix in the title to differenciate a new feature from a fix -->
<!-- Title format must be :
feat: FEATURE TITLE - MEED-XXXX - meeds-io/meeds#1234
or
fix: Fix TITLE - MEED-XXXX - meeds-io/meeds#1234
-->

<!-- Description : describe the feature/the fix by answering theses questions : -->
<!-- Why is this change needed?-->
<!-- Prior to this change, ...-->
<!-- How does it address the issue?-->
<!-- This change ...-->


<!-- Tips : 
Try To Limit Each Line to a Maximum Of 72 Characters
Provide links or keys to any relevant tickets, articles or other resources

Remember to
- Capitalize the subject line
- Use the imperative mood in the subject line
- Do not end the subject line with a period
- Separate subject from body with a blank line
- Use the body to explain what and why vs. how
- Can use multiple lines with "-" for bullet points in body
-->

rdenarie committed:
feat: Create page login in global site - MEED-9372 - Meeds-io/MIPs#203

Add css helper for full window height and width

rdenarie committed:
fix: when sticky is active on section, the left column have a top padding - MEED-9764 - meeds-io/MIPs#203

Before this fix, when a section is marked as sticky, there is a top of 13px which is displayed

rdenarie committed:
fix: Use css helper to set background-color for login buttons - MEED-9765 - meeds-io/MIPs#203

Before this fix, background color for login button are set with style directive
This modification use a css helper instead

rdenarie committed:
fix: update input text border color to improve constrast - MEED-9784 - meeds-io/MIPs#203

Before this fix, the border color for text input was too low in constrast.
This commit update the color to greyColorLighten1 (#707070) in order to improve this contrast

boubaker commented on platform-ui-skin/src/main/webapp/skin/less/core/helpers.less:
```diff
@@ -474,6 +474,14 @@
 }
+.full-window-height {
+  height: 100vh !important;
+  min-height: 100vh !important;
```
Use `calc(var(--100vh, 100vh)` instead.  See https://github.com/Meeds-io/social/commit/b13167749ae9e43914b01ce32a31e8209b2df75c#diff-744ec850ed3e5d90bca712e73671eae896e111af7d78802412c95ff5eefa3acf

> This change will allow to define a custom value for viewport definition to workaround the problem of viewport real size in Mobile Devices.

rdenarie replied:
done

boubaker replied:
Please do the same for the attribute `height: 100vh !important`

rdenarie replied:
done

boubaker replied:
<img width="499" height="119" alt="image" src="https://github.com/user-attachments/assets/4ba70943-c7ab-4c67-bcde-47ac78e0657a" />

rdenarie replied:
sorry, I miss my commit.
Should be ok now

boubaker commented on platform-ui-skin/src/main/webapp/skin/less/core/helpers.less:
```diff
@@ -1152,3 +1160,8 @@
   border: 1px solid @primaryBackground !important;
 }
+
+.white-background {
+  background-color: #FFFFFF !important;
+}
```
Why not using the `white` class ?

rdenarie replied:
white class also bring white border
https://github.com/Meeds-io/platform-ui/blob/develop/platform-ui-skin/src/main/webapp/skin/less/vuetify/lib/vuetify.less#L354-L357

And I need only white backgorund

boubaker replied:
ok :+1:

rdenarie committed:
fix : Correct feedbacks for merge - Meeds-io/MIPs#203 - MEED-9791

- Update button Signin with Email in loginForm
- This commit add a helper to display a button on multiple lines

boubaker commented on platform-ui-skin/src/main/webapp/skin/less/core/helpers.less:
```diff
@@ -474,6 +474,14 @@
   min-width: 100% !important;
 }
+.full-window-height {
+  height: 100vh !important;
+  min-height: calc(var(--100vh), 100vh) !important;
```
```suggestion
  height: ~"calc(var(--100vh), 100vh) !important";
  min-height: ~"calc(var(--100vh), 100vh) !important";
```

boubaker commented on platform-ui-skin/src/main/webapp/skin/less/core/helpers.less:
```diff
@@ -474,6 +474,14 @@
   min-width: 100% !important;
 }
+.full-window-height {
+  height: calc(var(--100vh), 100vh) !important;
+  min-height: calc(var(--100vh), 100vh) !important;
```
The usage of `~"..."` is to ensure that `calc` isn't interpreted as LESS command but CSS.

```suggestion
  height: ~"var(--100vh, 100vh) !important";
  min-height: ~"var(--100vh, 100vh) !important";
```

rdenarie replied:
done
